Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.py

```diff
@@ -0,0 +1,188 @@
+    :param raw: When set to "true" will return only rawJSON
+    :param limit: upper limit of indicators to fetch
+    :param (Optional) added_after: added after time string in parse_date_range format
+    :return: indicators in cortex TIM format
+    """
+    limit = try_parse_integer(limit)
+    if added_after:
+        added_after, _ = parse_date_range(added_after, date_format=TAXII_TIME_FORMAT)
+    raw = raw == "true"
+
+    indicators = client.build_iterator(limit=limit, added_after=added_after)
+
+    if raw:
+        demisto.results({"indicators": [x.get("rawJSON") for x in indicators]})
+        return
+
+    md = f"Found {len(indicators)} results:\n" + tableToMarkdown(
+        "", indicators, ["value", "type"]
+    )
+    if indicators:
+        return CommandResults(
+            outputs_prefix=CONTEXT_PREFIX + ".Indicators",
+            outputs_key_field="value",
+            outputs=indicators,
+            readable_output=md,
+        )
+    return md
+
+
+def get_collections_command(client):
+    """
+    Get the available collections in the TAXII server
+    :param client: FeedClient
+    :return: available collections
+    """
+    collections = list()
+    for collection in client.collections:
+        collections.append({"Name": collection.title, "ID": collection.id})
+    md = tableToMarkdown("TAXII2 Server Collections:", collections, ["Name", "ID"])
+    return CommandResults(
+        outputs_prefix=CONTEXT_PREFIX + ".Collections",
+        outputs_key_field="ID",
+        outputs=collections,
+        readable_output=md,
+    )
+
+
+def reset_fetch_command(client):
+    """
+    Reset the last fetch from the integration context
+    """
+    demisto.setIntegrationContext({})
+    return (
+        "Fetch was reset successfully. Your next indicator fetch will collect indicators from "
+        'the configured "First Fetch Time"'
+    )
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    url = params.get("url")
+    collection_to_fetch = params.get("collection_to_fetch")
+    credentials = params.get("credentials") or {}
+    username = credentials.get("identifier")
+    password = credentials.get("password")
+    proxies = handle_proxy()
+    verify_certificate = not params.get("insecure", False)
+
+    command = demisto.command()
+    demisto.info(f"Command being called is {command}")
+
+    try:
+        client = Taxii2FeedClient(
+            url, collection_to_fetch, proxies, verify_certificate, username, password
+        )
+        client.initialise()
+        commands = {
+            "taxii2-reset-fetch-indicators": reset_fetch_command,
+            "taxii2-get-indicators": get_indicators_command,
+            "taxii2-get-collections": get_collections_command,
+        }
+
+        if demisto.command() == "test-module":
+            # This is the call made when pressing the integration Test button.
+            test_module(client)
+
+        elif demisto.command() == "fetch-indicators":
```

Comment:
let's add a message in debug about the len of IOC fetched, the time, etc.

Reply:
I do it as part of `extract_indicators_from_envelope_and_parse` since I have all the data there ready. Do you think I should print it here?

Reply:
no need.